src/main/scala/it/pagopa/interop/catalogprocess/common/readmodel/ReadModelCatalogQueries.scala

```diff
@@ -193,8 +208,22 @@ object ReadModelCatalogQueries extends ReadModelQuery {
 
     val modeFilter = mode.map(_.toString).map(Filters.eq("data.mode", _))
 
+    val visibilityRestrictionsFilter =
+      if (visibilityRestrictions)
+        Some(
```

Comment:
awesome! 🚀 

I think it requires also a filter based on the role.
For example using something like the `visibilityRestrictions` flag (calculated from the roles) in the query to allow ADMIN and API producers to see the eservice, but not other roles (SECURITY, SUPPORT, ...), even if they belong to the producer.

May I also ask to add a comment to explain the logic?
Something like
```
Allow only the producer with right permissions to see eservices in draft, that is:
- eservices without descriptors
- eservices with only one descriptor that is in Draft state
```